### PR TITLE
[sdkgen] Add test schema with aliases

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
Previously we didn't have a test schema with aliases in SDK-gen as shown by the code coverage report, this PR fixes that 😄 

## Checklist

- [ ] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/codegen/testing/test/testdata/simple-resource-with-aliases/schema.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        "bar": {
+          "type": "string"
+        }
+      },
+      "requiredInputs": ["bar"],
+      "type": "object",
+      "aliases": [
+        {
+          "type": "example:index:BasicResource",
+          "name": "BasicResource",
```
Comment (Frassle) [inline pkg/codegen/testing/test/testdata/simple-resource-with-aliases/schema.json:37]: Lol really? We let sdkgen set Name aliases? That's so broken. The only thing an SDK alias should set is the type.

Comment (Zaid-Ajaj) [reply pkg/codegen/testing/test/testdata/simple-resource-with-aliases/schema.json:37]: Oh so these should be removed from sdk-gen 🤔 I've looked through the usages of `name` and `project` from `*scheme.Alias` and its only in sdk-gen. Should we remove `name` and `project` from the schema entirely?

Comment (Frassle) [reply pkg/codegen/testing/test/testdata/simple-resource-with-aliases/schema.json:37]: Yeh, only Type makes sense for SDKs. Later we could fix the schema to just be a string list (although we'd have to suppport `{"type": "str"}` list as well for compat.

**Comment by Zaid-Ajaj** [comment]
Closing in favor of #14625 where we removed writing out `name` and `project` of aliases and re-added this test schema

**Pull request closed without merging**